src/library/waveform.js

```diff
@@ -0,0 +1,57 @@
+// This is the library for creating the waveform
+class Waveform {
+
+  /** This is the constructor for initializing waveform
+   * @param {String} canvasId The HTML id where waveform is located.
+   */
+  constructor(canvasId) {
+    this.canvas = document.querySelector(canvasId);
+    this.canvasContext = this.canvas.getContext('2d');
+    this.width = this.canvas.width;
+    this.height = this.canvas.height;
+    this.currentX = 0;
+    this.previousY = this.height / 2;
+  }
+
+  /** This is the create function for creating waveform
+   * @param {AnalyserNode} analyserNode The analysis node
+   *     which connect with the audio context.
+   */
+  create(analyserNode) {
+    analyserNode.fftSize = 32;
```

Comment:
Why 32?

Reply:
This is the minimum fftSize which we are able to have. By using size 32, we can collect the most accurate data.